pkg/driver/aws-ebs/aws_ebs_tags_retry_worker.go

```diff
@@ -0,0 +1,121 @@
+package aws_ebs
+
+import (
+	"context"
+	"errors"
+	configv1 "github.com/openshift/api/config/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+)
+
+// startFailedQueueWorker runs a worker that processes failed volumes independently
+func (c *EBSVolumeTagsController) startFailedQueueWorker(ctx context.Context, syncContext factory.SyncContext) error {
+	for {
+		select {
+		case <-ctx.Done():
+			klog.Infof("Context canceled, stopping failed queue worker for EBS Volume Tags")
+			return errors.New("context canceled, stopping failed queue worker for EBS Volume Tags")
+		default:
+			item, quit := c.failedQueue.Get()
+			if quit {
+				klog.Infof("Failed queue worker is shutting down")
+				return errors.New("failed queue worker is shutting down")
+			}
+			c.processFailedVolume(ctx, item)
+		}
+	}
+}
+
+// processFailedVolume processes a single failed volume from the queue
+func (c *EBSVolumeTagsController) processFailedVolume(ctx context.Context, obj interface{}) {
+	defer c.failedQueue.Done(obj)
+	pvName, ok := obj.(string)
+	if !ok {
+		klog.Errorf("Invalid volume name type in failed queue, skipping")
+		c.failedQueue.Forget(obj)
+		return
+	}
+
+	klog.Infof("Retrying failed volume: %v", pvName)
+
+	infra, err := c.getInfrastructure()
```

Comment:
Race condition: As I understand from @anirudhAgniRedhat , the worker is a separate thread running concurrently. In that case, the call here does not gurantee that the infrastructure being read is latest. There is probability of older version infrastructure tags and updating to the pv. As this operation is not iterated until next update, it will lead to a situation where some of the volumes carry older versions of tags.

Reply:
Correct maybe we would need to resync in every x mins, if the volume is out of sync due to reading older data here.

WDYT @gnufied @jsafrane ?